org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Scanner.java

```diff
@@ -575,230 +600,23 @@ protected final boolean scanForTextBlockClose() throws InvalidInputException {
 	return false;
 }
 public char[] getCurrentTextBlock() {
-	// 1. Normalize, i.e. convert all CR CRLF to LF
+	char[][] lines = getCurrentTextBlockAsLines();
+	int indent = TextBlock.getTextBlockIndent(lines);
+	//	get rid of all the cached values
+	this.textBlockOffset = -1;
+	return TextBlock.formatTextBlock(lines, indent);
```

Comment:
Clearing of cached values is better handled elsewhere. After getNextToken() returns a template, it should be possible to call getCurrentTextBlock() multiple times for whatever reason